.github/workflows/ci.yml

```
@@ -13,17 +13,25 @@ jobs:
         os: [macos-12, ubuntu-20.04]
     runs-on: ${{ matrix.os }}
     steps:
-      - uses: actions/checkout@v2
-      - run: swift build --disable-automatic-resolution -c release
+        - uses: actions/checkout@v4
+        - uses: fwal/setup-swift@v2
+        - name: Echo Swift version
+          run: swift --version
+        - name: Check a Release build
+          run: swift build --disable-automatic-resolution -c release
 
   run-tests:
     strategy:
       matrix:
-        swift-version: [5.6, 5.7, 5.8, 5.9]
+        # We don't need anything specific in 5.10.1, but 5.10 collapses to 5.1 when `swift-setup` runs.
+        swift-version: [5.6, 5.7, 5.8, 5.9, 5.10.1]
```

We should quote the values to avoid that issue:

```suggestion
        swift-version: ['5.6', '5.7', '5.8', '5.9', '5.10']
```